packages/compass-web/polyfills/http/index.ts

```diff
@@ -0,0 +1,17 @@
+import { EventEmitter } from 'events';
+class Server extends EventEmitter {
+  listen() {
+    queueMicrotask(() => {
+      this.emit('listening');
+    });
+  }
+  address() {
+    return null;
+  }
+  close() {
+    // noop
+  }
+}
+export function createServer() {
```

Comment:
Why is webpack encountering this?

Reply:
Don't know if I'm phrasing these questions correctly ;)

Reply:
I think this is also covered by https://github.com/mongodb-js/compass/pull/5249#discussion_r1434288374, I'll add that in particular this `http` polyfill here is needed because currently when starting a data-service connection process, oidc plugin inside devtools-connect package will try to start its http server that handles auth flow and will not proceed until server is ready, this is why the polyfill is a bit more involved than others here